### PR TITLE
[go] Build go1.17.5 and go1.16.12 images

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature
/area dependency

#### What this PR does / why we need it:
Build image using go1.17.5 and go1.16.12

/assign @puerco @saschagrunert @Verolop @justaugustus @xmudrii 
/cc @kubernetes/release-engineering 

#### Which issue(s) this PR fixes:
Part of #2349 

#### Does this PR introduce a user-facing change?

```release-note
[go] Build go1.17.5 and go1.16.12 images
```

### CODE REVIEW

File: images/build/cross/Makefile [hunk 1]
```diff
@@ -27,8 +27,8 @@ IMGNAME = kube-cross
 # Example:
 # - v1.100.0-go1.17-bullseye.0 satisfies SemVer regex, while:
 # - v1.100-go1.17-bullseye.0 does not
-KUBERNETES_VERSION ?= v1.22.0
-GO_VERSION ?= 1.17.4
+KUBERNETES_VERSION ?= v1.23.0
```
Comment (cpanato) [inline images/build/cross/Makefile:30]: updated here as well to the newest release, not sure if that is ok, just to confirm

**Review by justaugustus — APPROVED** [review]
/area release-eng/security

> updated here as well to the newest release, not sure if that is ok, just to confirm

@cpanato -- Yep, that's fine!

File: images/build/cross/Makefile (hunk 1 above)
Comment (justaugustus) [reply images/build/cross/Makefile:30]: https://github.com/kubernetes/release/pull/2351#pullrequestreview-828795810

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

**Pull request merged**